cmd/buildtools/utils.go

```diff
@@ -11,12 +12,90 @@ import (
 	"os"
 	"path/filepath"
 	"regexp"
+	"sort"
+	"strconv"
 	"strings"
 
 	"github.com/google/go-github/v62/github"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	wolfiAPKIndexURL = "https://packages.wolfi.dev/os/x86_64/APKINDEX.tar.gz"
+)
+
+func GetWolfiAPKIndex() ([]byte, error) {
+	tmpdir, err := os.MkdirTemp("", "wolfi-apk-index")
+	if err != nil {
+		return nil, fmt.Errorf("create temp dir: %w", err)
+	}
+	defer os.RemoveAll(tmpdir)
+	if err := DownloadFile(wolfiAPKIndexURL, filepath.Join(tmpdir, "APKINDEX.tar.gz")); err != nil {
+		return nil, fmt.Errorf("download APKINDEX.tar.gz: %w", err)
+	}
+	if err := ExtractTGZArchive(filepath.Join(tmpdir, "APKINDEX.tar.gz"), tmpdir); err != nil {
+		return nil, fmt.Errorf("extract APKINDEX.tar.gz: %w", err)
+	}
+	contents, err := os.ReadFile(filepath.Join(tmpdir, "APKINDEX"))
+	if err != nil {
+		return nil, fmt.Errorf("read APKINDEX: %w", err)
+	}
+	return contents, nil
+}
+
+func GetWolfiPackageVersion(wolfiAPKIndex []byte, pkgName string, pinnedVersion string) (string, error) {
+	// example APKINDEX content:
+	// P:calico-node
+	// V:3.26.1-r1
+	// ...
+	//
+	// P:calico-node
+	// V:3.26.1-r10
+	// ...
+	//
+	// P:calico-node
+	// V:3.26.1-r9
+	// ...
+
+	var revisions []int
+	scanner := bufio.NewScanner(bytes.NewReader(wolfiAPKIndex))
+	for scanner.Scan() {
+		line := scanner.Text()
+		// filter by package name
+		if !strings.HasPrefix(line, "P:"+pkgName) {
```

Comment:
what if the package is a prefix of another package? like `P:prometheus` and `P:prometheus-adapter`? why not match the entire line?

Reply:
I was just thinking about that. I'll fix it.